### PR TITLE
fix(runtime): use object mapper to convert to map

### LINKED ISSUE #135: HTTP connector in `camunda/connectors:0.2.0-rc1-ootb` image fails to evaluate response mapping expression
__Describe the Bug__

I'm trying to run the ootb connectors via `camunda/connectors:0.2.0-rc1-ootb`.

Using the rest connector with a result mapping fails:

```
Sep 29, 2022 10:51:16 AM io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler handle
INFO: Received job 2251799813851702
Sep 29, 2022 10:51:16 AM io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler handle
SEVERE: Failed to process job 2251799813851702
io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapperException: Failed to evaluate expression '=result.body.id' and context 'HttpJsonResult{status=200, headers={date=Thu, 29 Sep 2022 10:51:16 GMT, transfer-encoding=chunked, keep-alive=timeout=60, content-type=application/vnd.spring-boot.actuator.v3+json, connection=keep-alive}, body={id=310}}', because expression evaluation failed
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:84)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.lambda$createOutputVariables$0(ConnectorJobHandler.java:97)
	at java.base/java.util.Optional.map(Unknown Source)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.createOutputVariables(ConnectorJobHandler.java:97)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.handle(ConnectorJobHandler.java:73)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class io.camunda.connector.http.model.HttpJsonResult cannot be cast to class java.util.Map (io.camunda.connector.http.model.HttpJsonResult is in unnamed module of loader 'app'; java.util.Map is in module java.base of loader 'bootstrap')
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.ensureVariablesMap(FeelEngineWrapper.java:70)
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:75)
	... 12 more
```

The expression I used is `= result.body.id`. In the exception we can see that this variable is available.

__Steps to Reproduce__

1. Deploy `camunda/connectors:0.2.0-rc1-ootb`
2. Use the HTTP connector with a response mapping expression

__Expected Behavior__

The HTTP connector in the OOTB docker image can evaulate response mapping expression.

### PR DESCRIPTION
**Pull request by menski**
## Description

To support POJO responses of the connector function use the jackson object mapper to ensure that response can be casted to a map.

## Related issues

closes #135

### CODE REVIEW

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java [hunk 1]
```diff
@@ -31,6 +32,8 @@ public class FeelEngineWrapper {
   static final String RESPONSE_MAP_KEY = "response";
   static final String ERROR_VARIABLES_MUST_NOT_BE_NULL = "variables cannot be null";
   static final String ERROR_EXPRESSION_EVALUATION_FAILED = "expression evaluation failed";
+  public static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE =
```
Comment (nikku) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java:35]: Why shall this be public?

File: runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java [hunk 2]
```diff
@@ -114,6 +114,25 @@ public void shouldSetToResultExpression() {
...
+    @Test
+    public void shouldSetToResultExpressionWhenPojoIsReturned() {
+      // given
+      // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
+      final String responseValue = "response";
+      var jobHandler =
+          new ConnectorJobHandler((context) -> new TestConnectorResponsePojo(responseValue));
+
+      // FEEL expression -> {"processedOutput":response.callStatus}
+      final String resultExpression = "{\"processedOutput\": response.value }";
```
Comment (nikku) [inline runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java:126]: We could also assert that we do not choke on `FEEL expression -> = response`.

**Review by nikku — CHANGES_REQUESTED** [review]
Proposal: As we touch this let us go the extra mile and cover all possible cases, at best gracefully:

```
= result
= result.body
```

Still need to gracefully intercept mapping of scalar types:

```
= result.body.someString
```

Cf. https://github.com/camunda/connector-sdk/issues/145#issue-1396257553

File: runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java [hunk 3]
```diff
@@ -114,6 +114,25 @@ public void shouldSetToResultExpression() {
           .isEqualTo(Map.of("processedOutput", Map.of("statusCode", "200 OK")));
     }
 
+    @Test
+    public void shouldSetToResultExpressionWhenPojoIsReturned() {
+      // given
+      // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
```
Comment (igpetrov) [inline runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java:120]:
```suggestion
      // Response from service -> POJO-object
```

**Comment by nikku** [comment]
@menski Please checkout https://github.com/camunda/connector-sdk/pull/148 where I implemented variable mappings + improved error messages in the mentioned cases.

**Comment by tmetzke** [comment]
Thanks a lot for noticing and fixing this, @menski !! One thing that came to my mind here: shouldn't we also convert POJOs for the result variable case? Right now, we're just handling the result expression use case. Do we need to serialize and deserialize the connector result? Does that make sense? This is what happens in the SaaS setup, right?

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java (hunk 1 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java]: Fixed via https://github.com/camunda/connector-sdk/pull/146/commits/61fbbaa49d2f6a57beab969d423bff3155fa23ea.

**Comment by nikku** [comment]
> One thing that came to my mind here: shouldn't we also convert POJOs for the result variable case? Right now, we're just handling the result expression use case.

@tmetzke Variables passed to the Job completion command will be serialized to JSON using a built-in object mapper. Thus I don't believe this is an issue.

**Review by nikku — APPROVED** [review]
Added https://github.com/camunda/connector-sdk/pull/146/commits/85b663fb95087a67c22563f6979abc6dee9fb96d as a dedicated test for the FEEL engine wrapper.

Let's handle everything else as a follow-up.

**Comment by nikku** [comment]
General observation: Our tests can get an overhaul / improved naming / structure. To be handled outside of this PR.

File: runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java (hunk 3 above)
Comment (nikku) [reply runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java:120]: Relatest to https://github.com/camunda/connector-sdk/pull/146#issuecomment-1268029311.

I'll take this clean-up as a follow up, as the dust settles.

**Pull request merged**